src/main/java/org/prebid/server/cache/CacheService.java

```diff
@@ -391,19 +393,23 @@ private static PutObject createPutObjectVideoOnly(Bid bid) {
 
     /**
      * Makes JSON type {@link PutObject} from {@link com.iab.openrtb.response.Bid}.
-     * Used for OpenRTB auction request. Also, adds win url to result object.
+     * Used for OpenRTB auction request. Also, adds win url to result object if events are enabled.
      */
     private PutObject createJsonPutObjectOpenrtb(CacheBid cacheBid, Map<String, List<String>> biddersToCacheBidIds,
-                                                 String accountId, Long timestamp) {
+                                                 Account account, boolean eventsAllowedByRequest, Long timestamp) {
         final com.iab.openrtb.response.Bid bid = cacheBid.getBid();
         final String bidId = bid.getId();
+
         final ObjectNode bidObjectNode = mapper.mapper().valueToTree(bid);
-        biddersToCacheBidIds.entrySet().stream()
-                .filter(biddersAndBidIds -> biddersAndBidIds.getValue().contains(bidId))
-                .findFirst()
-                .map(Map.Entry::getKey)
-                .ifPresent(bidder -> bidObjectNode.put("wurl", eventsService.winUrl(bidId, bidder, accountId,
-                        timestamp)));
+        final boolean eventsEnabled = account.getEventsEnabled() != null;
+        if (eventsEnabled && eventsAllowedByRequest) {
```

Comment:
According to requirements `eventsAllowedByRequest` should not be applied to `CacheService` at all.